src/components/grid/grid.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import customProperties from '../../../materials/custom-properties.json';
+import filterDataAttributes from '../../utils/filter-data-attributes';
+
+// TODO: check with design if those "breakpoints" are ok
+const minColumnSizes = {
+  1: '600px',
+  2: '450px',
+  3: '350px',
+  4: '275px',
+};
+
+const GridContainer = styled.div`
+  display: grid;
+  grid-gap: ${customProperties['--spacing-16']};
+  grid-auto-columns: 1fr;
+  grid-template-columns: ${props => {
+    switch (props.mode) {
```

Comment:
Wondering if a `Grid.Fixed` and `Grid.AutoFit` would be nicer than a `prop`. It feels a bit like a `type` on a button. I know it would likely complicate the implementation a bit cause we'd have the `GridContainer` and compose it with more specific `grid-template-columns` depending if it's `Grid.Fixed` and `Grid.Autofit`. Not sure if the suggestion makes sense or if others would agree.

Reply:
Sure, we can do that. It would avoid "throwing" errors because of unsupported modes.

Reply:
We could also make one mode the default instead of throwing.

Reply:
https://github.com/commercetools/ui-kit/pull/442/commits/f9ac92b71af797caad5f13954bbefce89f75531d